### PR TITLE
fix: stop new rsvps cancelling attendance

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

The issue was that we sent requests to the Google api with `attendees: [ { email: 'an@add.ress'} ]` rather than `attendees: [ { email: 'an@add.ress'}, responseStatus: ... ]`, that meant Google would remove their previous `responseStatus`.

The fix was to first get the event data, modify it and then send the updated data.

Tests to follow in another PR.

### CODE REVIEW

File: server/src/services/Google.ts [hunk 1]
```diff
@@ -189,98 +185,166 @@ function getStandardPatchBody({
...
 
-  await calendarApi.events.patch({
-    calendarId,
-    eventId: calendarEventId,
-    sendUpdates: 'all',
-    requestBody: getStandardPatchBody(eventUpdateData),
-  });
+function filterFromAttendees(email: string) {
+  return (attendees: calendar_v3.Schema$EventAttendee[]) =>
+    attendees.filter((attendee) => attendee.email === email);
```
Comment (gikf) [inline server/src/services/Google.ts:285]: Are we leaving here other attendees, or just the one having `email`?
```suggestion
    attendees.filter((attendee) => attendee.email !== email);
```

**Review by gikf — COMMENTED** [review]
I like it very much in general.

File: server/src/services/Google.ts (hunk 1 above)
Comment (ojeytonwilliams) [reply server/src/services/Google.ts:285]: Oops. It's what you suggested.

Looks like I'm writing those tests now!

**Comment by ojeytonwilliams** [comment]
Phew! All done.

A few of the tests are a little trivial (does `deleteCalendarEvent` call `events.delete`? Yes. Yes it does), but overall I think they're reasonable.

I've not tested them locally, yet, I ran out of time. That's next on my list, but I figured it was worth sharing the code anyway.

File: server/tests/google.test.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        { attendeeEmail: 'b@person' },
+      );
+
+      expect(updatedEvent.data.attendees?.length).toBe(2);
+      expect(updatedEvent.data.attendees).toEqual(
+        arrayContaining(remainingAttendees),
+      );
+    });
+
+    it("should do nothing if the email one of the attendee's", async () => {
```
Comment (gikf) [inline server/tests/google.test.ts:116]:
```suggestion
    it("should do nothing if the email is not one of the attendee's", async () => {
```

File: server/src/util/calendar.ts [hunk 3]
```diff
@@ -2,64 +2,34 @@ import { inspect } from 'util';
...
-
 interface CreateCalendarEventData {
   attendeeEmails: string[];
   calendarId: string;
   event: Pick<Event, 'id' | 'ends_at' | 'start_at' | 'name'>;
 }
 
-export const createCalendarEvent = async ({
+// TODO: consider pulling this back into the resolver (it's only used twice,
+// it's not that complex so I don't think we need to abstract it out)q
```
Comment (Sboonny) [inline server/src/util/calendar.ts:15]:
```suggestion
// it's not that complex so I don't think we need to abstract it out)
```

**Review by Sboonny — APPROVED** [review]
I think I couldn't test it, because `responseStatus` did fail for me, but I have gone to the calendar and activated google auth. 🤔

Well, doesn't matter anymore.  👍 Thank you @gikf, for reviewing this. 💚

**Comment by ojeytonwilliams** [comment]
@Sboonny so we don't accidentally spam emails during Cypress testing, the attendees are set to `[]` when you're running the site in development mode.  Could it be that?

**Review by gikf — APPROVED** [review]
@ojeytonwilliams is there anything else you want to do here?

**Comment by Sboonny** [comment]
Oh I think, I am the one causing the delay, because the talk about not being able to test it

**Comment by ojeytonwilliams** [comment]
> is there anything else you want to do here?

Nope, I think it's fine.  I'll deploy it now and see if anything breaks.

**Pull request merged**